framework_tvb/tvb/tests/framework/adapters/uploaders/encrypt_decrypt_test.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+#
+#
+# TheVirtualBrain-Framework Package. This package holds all Data Management, and
+# Web-UI helpful to run brain-simulations. To use it, you also need do download
+# TheVirtualBrain-Scientific Package (for simulators). See content of the
+# documentation-folder for more details. See also http://www.thevirtualbrain.org
+#
+# (c) 2012-2020, Baycrest Centre for Geriatric Care ("Baycrest") and others
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this
+# program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+#   CITATION:
+# When using The Virtual Brain for scientific publications, please cite it as follows:
+#
+#   Paula Sanz Leon, Stuart A. Knock, M. Marmaduke Woodman, Lia Domide,
+#   Jochen Mersmann, Anthony R. McIntosh, Viktor Jirsa (2013)
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+#
+#
+
+"""
+.. moduleauthor:: Robert Vincze <robert.vincze@codemart.ro>
+"""
+
+import tvb_data
+import os
+import pyAesCrypt
+import pytest
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.backends import default_backend
+from tvb.adapters.uploaders.zip_connectivity_importer import ZIPConnectivityImporterModel
+from tvb.core.adapters.abcuploader import ABCUploader
+from tvb.interfaces.command.demos.importers.encrypt_data import encrypt_password, save_encrypted_password, \
+    ENCRYPTED_PASSWORD_NAME, get_path_to_encrypt, ENCRYPTED_DATA_SUFFIX, DECRYPTED_DATA_SUFFIX, generate_random_password
+from tvb.tests.framework.core.base_testcase import TransactionalTestCase
+from tvb.basic.profile import TvbProfile
+
+
+class TestEncryptionDecryption(TransactionalTestCase):
+
+    # noinspection PyTypeChecker
+    @pytest.mark.parametrize("dir_name, file_name", [('connectivity', 'connectivity_76.zip'),
+                                                     ('surfaceData', 'cortex_2x120k.zip'),
+                                                     ('projectionMatrix', 'projection_meg_276_surface_16k.npy'),
+                                                     ('h5', 'TimeSeriesRegion.h5')])
+    def test_encrypt_decrypt(self, dir_name, file_name):
```

Comment:
This runs surprisingly fast on Jenkins 
https://jenkins.codemart.ro/view/TVB/job/TVB%20-%20Tests%20-%20SqLite/869/testReport/adapters.uploaders.encrypt_decrypt_test/TestEncryptDecryptImporter/

Then your strange times for uploader execution (with/without encryption) could be due to other reason (e.g. machine load), as 0.x s for encrypt&decrypt in a ~10s job range would not be something to reasonably benchmark with one single manual test, but would require an automated setup to reliably measure multiple times.

Reply:
Yeah, I know that it runs surprisingly fast...